### ISSUE TITLE
Handle disallowed hashtags

### ISSUE THREAD

**Issue by connyduck** [tuskyapp/Tusky#665]
See https://github.com/tootsuite/mastodon/pull/7176

**Comment by charlag** [comment]
What should we do about them? Check that they exist when displaying them? I don't think the list of disallowed ones is exposed through the api

**Comment by connyduck** [comment]
I think the api returns a specific error when a disallowed hashtag is encountered. We should give the user this information so they know what to change.

**Comment by nikclayton** [comment]
tl;dr: This is almost done, but the error message is not shown to the user.

Longer:

If I'm reading the Mastodon code correctly, this doesn't return a specific HTTP error. Instead, it returns a generic error with *localised* text that tells you that the hashtag is not allowed (https://github.com/mastodon/mastodon/blob/main/app/validators/disallowed_hashtags_validator.rb#L8).

I can't imagine that trying to parse out the localised text from an error to figure out whether or not to display it is a sustainable solution.

So this would mean showing the error text of all `422` errors to the user.

Handily, I think this already almost happens.

`NetworkResult` treats 4xx responses as failures, and wraps them in `HttpException`.  This gets caught and sent to `failSending`, which checks if the failure is an `HttpException`. If it is, a notification with the failed status is created.

*But*, this notification does not contain the text of the error that was in the response from the Mastodon server.

So if the code that calls `failSending` was changed to include the error text, and if `failSending` included it in the notification, then I think this issue could be closed.
